### ISSUE TITLE
Req: Easy ROM reload for quick development cycles

### ISSUE THREAD

**Issue by nitro2k01** [tommitytom/RetroPlug#7]
I'm developing GB software that receives MIDI. I find RP to be incredibly useful for quick testing as it cuts out the cycle of burning a ROM to a flashcart, or developing some solution to fake MIDI data over the serial port. However, it's still slightly annoying that in order to reload ROM (as far as I can tell) you need to system, load ROM, and use the file dialog to select the same ROM file. It would be useful if there was a quicker way. Here are a couple of possible ways this could be achieved.

- Assign some simple way to reload the file through a GUI shortcut, for example double or triple clicking the window. Or use the topmost item in the right click menu (the one that shows the ROM name) to reload the ROM.
- Let the plugin look for changes to the loaded file and reload if it's rewritten.
- Make the process listen for some signal that can be triggered from a build script, for example listening to a TCP port and opening a connection to that ports reloads.

If for example double clicking the window would be considered possibly confusing for users, this might be a feature that needs to be enabled in the config file.

Thanks for the consideration.

**Comment by tommitytom** [comment]
Yep this sounds like a good idea!  I'll be adding in file watching at some point to deal with auto updating of custom samples for LSDJ, so it makes sense to wire that up for roms too.  However, until that's in I'll probably add a keyboard shortcut that can be configured via the config file.  I'm actually on holiday til the end of August so wont be able to add it til then!

**Comment by nitro2k01** [comment]
Personal preference would be make it accessible without keyboard because I use Renoise and would prefer to have keyboard enabled unchecked for the most part.

**Comment by tommitytom** [comment]
Yep, sure thing.  I'll add the file watcher in that case.  I plan to add Lua scripting next which will make all of this stuff a lot easier!
